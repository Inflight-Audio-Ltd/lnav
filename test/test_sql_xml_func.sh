#! /bin/bash

export YES_COLOR=1

run_cap_test ./drive_sql "SELECT * FROM xpath('/abc[', '<abc/>')"

run_cap_test ./drive_sql "SELECT * FROM xpath('/abc', '<abc')"

run_cap_test ./drive_sql "SELECT * FROM xpath('/abc/def', '<abc/>')"

run_cap_test ./drive_sql "SELECT * FROM xpath('/abc/def[@a=\"b\"]', '<abc><def/><def a=\"b\">ghi</def></abc>')"

run_cap_test ./drive_sql "SELECT * FROM xpath('/abc/def', '<abc><def>Hello &gt;</def></abc>')"

run_cap_test ${lnav_test} -n \
    -c ";SELECT * FROM xpath('/catalog', (SELECT content FROM lnav_file LIMIT 1))" \
    ${test_dir}/invalid-books.xml

run_cap_test ${lnav_test} -n \
    -c ";SELECT * FROM xpath('/cat[alog', (SELECT content FROM lnav_file LIMIT 1))" \
    ${test_dir}/books.xml

run_cap_test env TEST_XP1="//book[1]/price" TEST_XP2="//book[2]/price" \
    ${lnav_test} -n \
    -c ";SELECT * FROM environ, xpath(environ.value, (SELECT content FROM lnav_file)) WHERE name LIKE 'TEST_XP%'" \
    ${test_dir}/books.xml
