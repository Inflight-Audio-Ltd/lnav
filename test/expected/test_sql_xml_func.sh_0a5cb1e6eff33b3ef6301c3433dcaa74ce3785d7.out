[1m[4m  name  [0m[1m[4m [0m[1m[4m     value     [0m[1m[4m [0m[1m[4m       result        [0m[1m[4m [0m[1m[4m      node_path       [0m[1m[4m [0m[1m[4mnode_attr[0m[1m[4m [0m[1m[4m[7mnode_text [0m[1m[4m [0m
TEST_XP1 //book[1]/price <price>44.95</price>‚êä /catalog/book[1]/price {}        [7m     44.95[0m 
TEST_XP2 //book[2]/price <price>5.95</price>‚êä  /catalog/book[2]/price {}        [7m [0m     5.95 
